cmd/minishift/cmd/util/oc.go

```diff
@@ -92,3 +87,32 @@ func SetOcContext(profileName string) error {
 
 	return nil
 }
+
+//RemoveCurrentContext removes the current context from `machinename_kubeconfig`
+func RemoveCurrentContext() error {
+	ocPath := minishiftConfig.InstanceConfig.OcPath
+	cmd := "config unset current-context"
+
+	ocRunner, err := oc.NewOcRunner(ocPath, filepath.Join(constants.Minipath, "machines", constants.MachineName+"_kubeconfig"))
+	if err != nil {
+		if glog.V(2) {
+			fmt.Println(fmt.Sprintf("%s", err.Error()))
+		}
+		return errors.New(fmt.Sprintf("Error unsetting current-context"))
+	}
+
+	exitCode := ocRunner.RunAsUser(cmd, nil, nil)
+	if exitCode != 0 {
+		return errors.New(fmt.Sprintf("Error during removing current context: %v", err.Error()))
+	}
+	return nil
+}
+
+func GetOcPath(profileName string) string {
+	instanceConfigFile := filepath.Join(constants.GetProfileHomeDir(), "machines", profileName+".json")
+	instanceConfig, err := minishiftConfig.NewInstanceConfig(instanceConfigFile)
```

Comment:
This is wrong. NewInstanceConfig should be only called in root.go. We need to check if `minishiftConfig.InstanceConfig.OcPath` contains anything. If it does not exist that means the VM does not exist so we need to look for OcPath.

Reply:
You can just return `minishiftConfig.InstanceConfig.OcPath` if it exists or null string if it does not exist.

Reply:
the problem with returning `minishiftConfig.InstanceConfig.OcPath` is that, it gives the OcPath of the active profile, but while setting the OcContext we need the OcPath for the profile that we are setting the context to.

Reply:
@anjannath I understand that it is tricky as you need to get `InstanceConfig.OcPath` but is it not available for the profile you want to set as active.  I am working on the code change required without doing NewInstanceConfig() and will let you know. I do not want to run NewInstanceConfi() as it creates the file if the directory path is present.

Reply:
@LalatenduMohanty Sure, we don't want to create a file unnecessarily when the VM for that profile doesn't exists. But before calling `NewInstanceConfig` theres a check for the VM and it will fail if the VM doesn't exist, so we should never hit a situation where the file is not there and `NewInstanceConfig` creates it. But if we can do it without calling that function, that would be _the_ way to do it.